flang/lib/Optimizer/Transforms/OMPFunctionFiltering.cpp

```diff
@@ -66,6 +67,12 @@ class OMPFunctionFilteringPass
         SymbolTable::UseRange funcUses = *funcOp.getSymbolUses(op);
         for (SymbolTable::SymbolUse use : funcUses) {
           Operation *callOp = use.getUser();
+          if (mlir::isa<func::FuncOp>(callOp)) {
+            // Do not delete internal procedures holding the symbol of their
+            // Fortran host procedure as attribute.
+            callOp->removeAttr(fir::getHostSymbolAttrName());
```

Comment:
Can we remove the internal linkage attribute and set the visibility to public as well? Internal linkage and private visibility cause the offload function to be deleted. LLVM-Flang was able to compile and run the test case before https://github.com/llvm/llvm-project/pull/81929

Reply:
I updated to set the visibility to public, this is enough for MLIR not to remove it until the offload function is generated.